src/main/java/core/basesyntax/report/ReportGeneratorImpl.java

```diff
@@ -0,0 +1,24 @@
+package core.basesyntax.report;
+
+import core.basesyntax.service.ShopService;
+import java.util.Map;
+
+public class ReportGeneratorImpl implements ReportGenerator {
+    private ShopService shopService;
+
+    public void setShopService(ShopService shopService) {
+        this.shopService = shopService;
+    }
+
+    @Override
+    public String generateReport() {

```

Comment:
The `generateReport()` method is dependent on the `shopService.getFruits()` method. It would be better to pass the data to the `generateReport()` method as a parameter. This will make your `generateReport()` method more reusable and independent of `ShopService`. This is a code quality issue and a SOLID principle violation.